Core/Core/Planner/Model/API/GetPlannerNotesRequest.swift

```diff
@@ -0,0 +1,76 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+
+/// https://canvas.instructure.com/doc/api/planner.html#method.planner_notes.index
+public struct GetPlannerNotesRequest: APIRequestable {
+    public typealias Response = [APIPlannerNote]
+
+    public var path: String { "planner_notes" }
+
+    public let contexts: [Context]?
+    public let startDate: Date?
+    public let endDate: Date?
+    public let perPage: Int
+
+    public var useExtendedPercentEncoding: Bool { true }
+    private static let dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZZZZZ"
+        return formatter
+    }()
+
+    public init(
+        contexts: [Context]? = nil,
+        startDate: Date = Clock.now.addYears(-2),
+        endDate: Date = Clock.now.addYears(1),
+        perPage: Int = 100,
+        calendar: Calendar = .current,
+        timeZone: TimeZone = .current
+    ) {
+        self.contexts = contexts
+        self.startDate = startDate
+        self.endDate = endDate
+        Self.dateFormatter.calendar = calendar
+        Self.dateFormatter.timeZone = timeZone
```

Comment:
@rh12 
`GetPlannerNotesRequest` was actually written by copying `GetCalendarEventsRequest` then altering it to match the needed behavior for fetching planner notes. Therefore, for the common theme of code where you find such techniques, it was actually originated elsewhere in code.

But I agree with you on, it felt odd for me as well. The decision for leaving such examples, is that I was trying to replicate other patterns in code, so to move fast on this discovery work.

Reply:
One more thing, I understand the need to make this formatter as static. As it quite expensive to create one on each request. But I'd suggest persisting preferred calendar and timezone. Then set those on each time this formatter gets called. 

I will address that in code, and feel free to review it after that.

Reply:
Got it, thanks! Even though it's still the same case (mutable shared state), I'm fine with this, since this is a very contained thing, no need to improve it further.